dragonfly/story.py

```diff
@@ -0,0 +1,262 @@
+# coding: utf-8
+"""Dragonfly Story."""
+from ._base import _Base
+from .properties import StoryProperties
+from .roompolygon import RoomPolygon
+
+from honeybee.typing import float_positive
+from honeybee.model import Model
+
+from ladybug_geometry.geometry3d.polyface import Polyface3D
+
+
+class Story(_Base):
+    """A Story of a building defined by an extruded FloorPolygons.
+
+    Properties:
+        * name
+        * display_name
+        * room_polygons
+        * floor_to_floor_height
+        * is_ground_floor
+        * is_top_floor
+        * parent
+        * has_parent
+        * geometry
+        * outline_curves
+        * floor_height
+        * volume
+        * floor_area
+        * exterior_wall_area
+        * exterior_aperture_area
+    """
+    __slots__ = ('_room_polygons', '_floor_to_floor_height', '_is_ground_floor',
+                 '_is_top_floor', '_tolerance', '_parent')
+
+    def __init__(self, name, room_polygons, floor_to_floor_height=None,
```

Comment:
From a design point of view how do you imagine to deal with solve adjacency between the rooms in the same story? Should you have a method for that in this class or do you expect one to solve adjacency before creating a story? Also how does add_room will affect that design.

Reply:
So there are a number of things that still need to be addressed in `solve_adjacency` workflows before we really start expanding them to other areas like this. First and foremost, we have not yet determined how someone would set all interior adjacencies to be adiabatic, or change them to be a custom e+ construction or rad material, or set up air mixing between adjacent zones with air walls. I want to get some of these answered by 11/8 but we don't need to figure out this question here by the deadline since the initial tests aren't planned to be with Room2D>Story>Building workflows (essentially all zones are going to be entire floors in the Dragonfly workflows planned for 11/8).

However, to let you know what I'm thinking now, I think we can support both the option of solving adjacencies beforehand and solving all adjacencies on the `Story`. But I feel we should not be automating the solving of adjacencies each time because it's a time intensive process and people should have the freedom to set the boundary conditions to what they want.

Reply:
Understood!

It might be helpful if you limit the options in Dragonfly and draw a line between what is possible and what is not. I don't know where that line will exactly be but having this in mind can help to focus on what makes Dargonfly different from Honeybee. Some of the more advanced features can only be available in Honeybee.